tests/foreman/api/test_docker.py

```diff
@@ -0,0 +1,680 @@
+        ).create_json()['id']
+
+        repo_id = _create_repository(prod_id, name)['id']
+
+        # Create content view and associate docker repo
+        content_view = entities.ContentView(
+            organization=self.org_id, composite=False
+        ).create_json()
+        _add_repo_to_content_view(repo_id, content_view['id'])
+
+        new_attrs = entities.ContentView(id=content_view['id']).read_json()
+        self.assertIn(repo_id, new_attrs['repository_ids'])
+
+        # Publish it...
+        self.assertTrue(_publish_content_view(content_view['id']))
+        # ... and grab its version ID (there should only be one version)
+        new_attrs = entities.ContentView(id=content_view['id']).read_json()
+        version_id = new_attrs['versions'][0]['id']
+
+        # Create composite content view and associate content view to
+        # it
+        comp_content_view = entities.ContentView(
+            organization=self.org_id, composite=True
+        ).create_json()
+        _add_content_view_to_composite_view(
+            comp_content_view['id'], version_id)
+
+        new_attrs = entities.ContentView(
+            id=comp_content_view['id']
+        ).read_json()
+        self.assertIn(version_id, new_attrs['component_ids'])
+
+    @run_only_on('sat')
+    def test_add_multiple_docker_repos_to_composite_content_view(self):
+        """@Test: Add multiple Docker-type repositories to a composite
+        content view.
+
+        @Assert: One repository is created with a Docker image and the
+        product is added to a random number of content views which are then
+        added to a composite content view.
+
+        @Feature: Docker
+
+        """
+        prod_ids = []
+        cv_version_ids = []
+
+        for _ in range(0, randint(1, 5)):
+            prod_ids.append(
+                entities.Product(organization=self.org_id).create_json()['id']
+            )
+
+        for prod_id in prod_ids:
+            repo_id = _create_repository(prod_id)['id']
+
+            # Create content view and associate docker repo
+            content_view = entities.ContentView(
+                organization=self.org_id, composite=False
+            ).create_json()
+            _add_repo_to_content_view(repo_id, content_view['id'])
+
+            new_attrs = entities.ContentView(id=content_view['id']).read_json()
+            self.assertIn(repo_id, new_attrs['repository_ids'])
+
+            # Publish it...
+            self.assertTrue(_publish_content_view(content_view['id']))
+            # ... and grab its version ID (there should only be one version)
+            new_attrs = entities.ContentView(id=content_view['id']).read_json()
+            cv_version_ids.append(new_attrs['versions'][0]['id'])
+
+        # Create composite content view and associate content view to
+        # it
+        comp_content_view = entities.ContentView(
+            organization=self.org_id, composite=True).create_json()
```

Comment:
I think `id` is the only attribute accessed in this dict. Can you do something like this?

    comp_content_view_id = entities.ContentView(
        ...
    ).create_json()['id']